ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/electra/helpers/BeaconStateMutatorsElectra.java

```diff
@@ -193,6 +200,33 @@ public void switchToCompoundingValidator(final MutableBeaconStateElectra state,
     }
   }
 
+  /**
+   * queue_excess_active_balance
+   *
+   * @param state beaconState
+   * @param validatorIndex validatorIndex
+   */
+  public void queueExcessActiveBalance(
+      final MutableBeaconStateElectra state, final int validatorIndex) {
+    final UInt64 balance = state.getBalances().get(validatorIndex).get();
+    final UInt64 minActivationBalance = specConfigElectra.getMinActivationBalance();
+
+    if (balance.isGreaterThan(minActivationBalance)) {
+      final UInt64 excessBalance = balance.minus(minActivationBalance);
+      state.getBalances().set(validatorIndex, SszUInt64.of(minActivationBalance));
+      final SszMutableList<PendingBalanceDeposit> pendingBalanceDeposits =
+          MutableBeaconStateElectra.required(state)
+              .getPendingBalanceDeposits()
+              .createWritableCopy();
+      pendingBalanceDeposits.append(
+          schemaDefinitionsElectra
+              .getPendingBalanceDepositSchema()
+              .create(
+                  SszUInt64.of(UInt64.fromLongBits(validatorIndex)), SszUInt64.of(excessBalance)));
+      state.setPendingBalanceDeposits(pendingBalanceDeposits);
```

Comment:
i *think* you can call append in place rather than having to repack the whole list...
```
          MutableBeaconStateElectra.required(state)
              .getPendingBalanceDeposits()
             .append(deposit)

```